packages/form-core/src/FormApi.ts

```diff
@@ -455,7 +517,13 @@ export class FormApi<
       }
 
       promises.push(
-        new Promise<ValidationError | undefined>(async (resolve) => {
+        new Promise<
+          | {
+              fieldErrors: Partial<Record<DeepKeys<TFormData>, ValidationError>>
+              errorMapKey: ValidationErrorMapKeys
+            }
+          | undefined
+        >(async (resolve) => {
```

Comment:
Could use `ValidationPromiseResult` here.

Reply:
Nice catch, thanks! ☺️ Fixed it!